src/Helper/Cart.php

```diff
@@ -0,0 +1,147 @@
+use Magento\Framework\Api\SimpleDataObjectConverter;
+use Shopgate\Base\Model\Config;
+use Shopgate\Base\Model\Utility\SgLoggerInterface;
+use Shopgate\Export\Helper\Quote as QuoteHelper;
+
+class Cart extends OriginalCartHelper
+{
+    /** @var Config */
+    private $config;
+    /** @var SgLoggerInterface */
+    private $logger;
+    /** @var QuoteHelper */
+    private $quoteHelper;
+    /** @var array */
+    private $quoteFields;
+    /** @var RequestInterface */
+    protected $request;
+    /** @var bool */
+    protected $isBoltRequest;
+
+    /**
+     * @param Config            $config
+     * @param SgLoggerInterface $logger
+     * @param QuoteHelper       $quoteHelper
+     * @param RequestInterface  $request
+     * @param array             $quoteFields
+     * @param array             $quoteStockFields
+     */
+    public function __construct(
+        Config $config,
+        SgLoggerInterface $logger,
+        QuoteHelper $quoteHelper,
+        RequestInterface $request,
+        array $quoteFields = [],
+        array $quoteStockFields = []
+    ) {
+        $this->config        = $config;
+        $this->logger        = $logger;
+        $this->quoteHelper   = $quoteHelper;
+        $this->quoteFields   = $quoteFields;
+        $this->request       = $request;
+        $this->isBoltRequest = $this->getBoltRequestFlag($request);
+        parent::__construct($config, $logger, $quoteHelper, $quoteFields, $quoteStockFields);
+    }
+
+    /**
+     * Takes in the allowed cart methods loaded up in the Base's DI
+     *
+     * @return array
+     * @throws \Exception
+     */
+    public function loadSupportedMethods()
+    {
+        $this->quoteHelper->load($this->quoteFields);
+        $fields = $this->loadMethods($this->config->getSupportedFieldsCheckCart());
+        if ($this->isBoltRequest) {
+            $this->quoteHelper->addBoltParentId();
+        }
+        if (!$this->isBoltRequest) {
+            $this->quoteHelper->cleanup();
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Loads the methods of the current class
+     *
+     * @param array $fields
+     *
+     * @return array
+     */
+    private function loadMethods(array $fields)
+    {
+        $methods = [];
+        foreach ($fields as $rawField) {
+            $method = 'get' . SimpleDataObjectConverter::snakeCaseToUpperCamelCase($rawField);
+            $this->logger->debug('Starting method ' . $method);
+            $methods[$rawField] = $this->{$method}();
+            $this->logger->debug('Finished method ' . $method);
+        }
+
+        return $methods;
+    }
+
+    /**
+     * @return mixed
+     */
+    protected function getInternalCartInfo()
+    {
+        $isRequestEmpty = empty($this->request) ? 'yes' : 'no';
+        $requestCart = $this->request->getParam('cart');
+        $internalCartInfo = $requestCart['internal_cart_info'];
+        if ($this->isBoltRequest) {
+            return json_encode([
```

Comment:
Magento requires you to use it's build in serializer: https://devdocs.magento.com/guides/v2.3/extension-dev-guide/framework/serializer.html#json-default instead of native php json_encode/json_decode

Reply:
moved over to the serializer